pkg/api/v1beta3/types.go

```diff
@@ -1010,7 +1009,7 @@ type ServicePort struct {
 	// If this is a string, it will be looked up as a named port in the
 	// target Pod's container ports.  If this is not specified, the value
 	// of Port is used (an identity map).
-	TargetPort util.IntOrString `json:"targetPort" description:"the port to access on the pods targeted by the service; defaults to the service port"`
+	TargetPort util.IntOrString `json:"targetPort,omitempty" description:"the port to access on the pods targeted by the service; defaults to the service port"`
```

Comment:
Another github annoyance -- I can't comment on arbitrary lines. :-(

ServiceAccount.Secrets should have omitempty:
https://github.com/nikhiljindal/kubernetes/blob/omitEmpty/pkg/api/v1beta3/types.go#L1052